File: Frontend/SoftSiniNet/SiniNetBusiness/PolizaVehicularBO.cs
using SiniNetBusiness.SiniNetWSPolizasVehiculares;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SiniNetBusiness
{
    public class PolizaVehicularBO
    {
        private PolizasVehicularesClient clienteSOAP;

        public PolizaVehicularBO()
        {
            this.clienteSOAP = new PolizasVehicularesClient();
        }

        public int insertar(
            personasDTO asegurado,
            estadoPoliza estado,
            tipoPolizaVehicular tipo,
            int porcentajeCobertura)
        {
            return this.clienteSOAP.insertarPolizasVehiculares(asegurado, estado, tipo, porcentajeCobertura);
        }

        public polizasVehicularesDTO obtenerPorId(int polizaId)
        {
            return this.clienteSOAP.obtenerPorIdPolizasVehiculares(polizaId);
        }

        public IList<polizasVehicularesDTO> listarTodos()
        {
            return this.clienteSOAP.listarTodosPolizasVehiculares();
        }

        public IList<polizasVehicularesDTO> listarPorNombreApellidoAsegurado(string filtro)
        {
            return this.clienteSOAP.listarPorNombreApellidoAseguradoPolizasVehiculares(filtro);
        }

        public int modificar(
            int polizaId,
            personasDTO asegurado,
            estadoPoliza estado,
            tipoPolizaVehicular tipo,
            int porcentajeCobertura)
        {
            return this.clienteSOAP.modificarPolizasVehiculares(polizaId, asegurado, estado, tipo, porcentajeCobertura);
        }

        public int eliminar(int polizaId)
        {
            return this.clienteSOAP.eliminarPolizasVehiculares(polizaId);
        }
    }

}

File: Backend/SoftAseg/SoftAsegBusiness/src/main/java/pe/edu/pucp/softaseg/bo/PolizaVehicularBO.java

package pe.edu.pucp.softaseg.bo;

import java.util.ArrayList;
import pe.edu.pucp.softaseg.dao.PolizaVehicularDAO;
import pe.edu.pucp.softaseg.daoImp.PolizaVehicularDAOImpl;
import pe.edu.pucp.softaseg.model.gestionSiniestros.PolizasVehicularesDTO;

public class PolizaVehicularBO {
    private PolizaVehicularDAO polizaVehicularDAO;
    
    public PolizaVehicularBO(){
        this.polizaVehicularDAO = new PolizaVehicularDAOImpl();
    }
    
    public Integer insertar(PolizasVehicularesDTO polizasVehicularesDTO){
        Integer polizaId = this.polizaVehicularDAO.insertar(polizasVehicularesDTO);
        polizasVehicularesDTO.setPolizaId(polizaId);
        return polizaId;
    }
    
    public PolizasVehicularesDTO obtenerPorId(Integer polizaId){
        PolizasVehicularesDTO polizasVehicularesDTO = new PolizasVehicularesDTO();
        polizasVehicularesDTO.setPolizaId(polizaId);
        return this.polizaVehicularDAO.obtenerPorId(polizaId);
    }
    
    public ArrayList<PolizasVehicularesDTO> listarTodos(){
        return this.polizaVehicularDAO.listarTodos();
    }
    
    public ArrayList<PolizasVehicularesDTO> listarPorNombreApellidoAsegurado(String filtro){
        return this.polizaVehicularDAO.listarPorNombreApellidoAsegurado(filtro);
    }
    
    public Integer modificar(PolizasVehicularesDTO polizasVehicularesDTO){
        return this.polizaVehicularDAO.modificar(polizasVehicularesDTO);
    }
    
    public Integer eliminar(Integer polizaId){
        PolizasVehicularesDTO polizasSOATDTO = new PolizasVehicularesDTO();
        polizasSOATDTO.setPolizaId(polizaId);
        return this.polizaVehicularDAO.eliminar(polizasSOATDTO);
    }
}
